### PR TITLE
[UI] Added Code Editor

### PR DESCRIPTION
**Pull request by meghanacosmos**
**Description**
Added Code Editor. 
Screenshot 2021-12-10 at 2 52 23 AM

**Notes for Reviewers**
This component accepts prop `defaultValue` which accepts a multiline string. I'm currently able to get the error row and the error in the console but I'm unable to show it on the editor.  

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -26,18 +26,26 @@
     "@fortawesome/free-brands-svg-icons": "^5.15.4",
     "@fortawesome/free-solid-svg-icons": "^5.15.4",
     "@fortawesome/react-fontawesome": "^0.1.16",
+    "@monaco-editor/react": "^4.3.1",
```
Comment (sudo-NithishKarthik) [inline ui/package.json:29]:
```suggestion

```

File: ui/package.json [hunk 2]
```diff
@@ -26,18 +26,26 @@
...
     "@mui/icons-material": "^5.0.0",
     "@mui/material": "^5.0.0",
     "@mui/styles": "^5.0.0",
     "@reduxjs/toolkit": "^1.6.1",
+    "ace-builds": "^1.4.13",
+    "brace": "^0.11.1",
```
Comment (sudo-NithishKarthik) [inline ui/package.json:35]: do we need this ?

File: ui/components/CodeEditor/CodeEditor.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          setValue(value);
+          checkerFunction(value);
+        }}
+        editorProps={{
+          $blockScrolling: true,
+        }}
+        fontSize={12}
+        value={value}
+        setOptions={{
+          useWorker: false,
```
Comment (sudo-NithishKarthik) [inline ui/components/CodeEditor/CodeEditor.js:44]: interesting 🤔 , we might spin up a web worker if we have to parse huge files.

File: ui/package.json [hunk 4]
```diff
@@ -26,18 +26,26 @@
...
     "classnames": "^2.2.6",
     "detect-port": "^1.3.0",
     "eslint-plugin-jest": "^24.4.2",
+    "file-loader": "^6.2.0",
```
Comment (sudo-NithishKarthik) [inline ui/package.json:39]: is this needed ? if not, will you please remove this ?

File: ui/package.json [hunk 5]
```diff
@@ -26,18 +26,26 @@
...
     "isomorphic-unfetch": "^3.1.0",
+    "js-yaml": "^4.1.0",
     "msw": "^0.35.0",
     "next": "^12.0.2",
-    "notistack": "2.0.3",
+    "next-dynamic-loading-props": "^0.1.1",
+    "notistack": "^2.0.3",
     "react": "^17.0.2",
+    "react-ace": "^9.5.0",
+    "react-codemirror2-react-17": "^1.0.0",
```
Comment (sudo-NithishKarthik) [inline ui/package.json:48]: Is this needed as well ?

**Review by sudo-NithishKarthik — CHANGES_REQUESTED** [review]
Good job @Meghana-12 🎉 . Just some small issues to address.

**Comment by sudo-NithishKarthik** [comment]
Also, can you set the DCO to pass ? 😅

File: ui/package.json (hunk 2 above)
Comment (Meghana-12) [reply ui/package.json:35]: Ace-builds is needed. I'll remove the` brace` dependency.

File: ui/components/CodeEditor/CodeEditor.js (hunk 3 above)
Comment (Meghana-12) [reply ui/components/CodeEditor/CodeEditor.js:44]: I had to set that fix some errors I think. I don't remember properly. But yeah, I think we can look into it later when we need to parse big files.

Comment (sudo-NithishKarthik) [reply ui/components/CodeEditor/CodeEditor.js:44]: Yep

**Review by sudo-NithishKarthik — APPROVED** [review]

**Pull request merged**